gtsam/basis/Basis.h

```diff
@@ -0,0 +1,411 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file Basis.h
+ * @brief Compute an interpolating basis
+ * @author Varun Agrawal, Frank Dellaert
+ * @date July 4, 2020
+ */
+
+#pragma once
+
+#include <gtsam/base/Matrix.h>
+#include <gtsam/base/OptionalJacobian.h>
+#include <gtsam/basis/ParameterMatrix.h>
+
+#include <iostream>
+#include <unsupported/Eigen/KroneckerProduct>
+
+/**
+ * This file supports creating continuous functions `f(x;p)` as a linear
+ * combination of `basis functions` such as the Fourier basis on SO(2) or a set
+ * of Chebyshev polynomials on [-1,1].
+ * In the expression `f(x;p)` the variable `x` is
+ * the continuous argument at which teh function is evaluated, and `p` are
+ * parameters, e.g., coefficients used to weight the different basis functions.
+ * However, different parameterizations are also possible.
```

Comment:
Isn't this numerical analysis 101 stuff? I may be wrong since I've only ever sat in on one numerical analysis course.

Reply:
Not sure - either way, JD and Michael use this code and it took us all ~an hour of reading through this code together to finally figure out what was going on :)   (this was prior to the last review cycle, it's much better now but still)

Reading through it a second time last night it still took me ~5 minutes to wrap my head around (no fault of your own, just I wasn't sure what "Weights" was referring to and I didn't see a literature reference.  I looked on [wikipedia](https://en.wikipedia.org/wiki/Chebyshev_polynomials) and [wolfram mathworld](https://mathworld.wolfram.com/ChebyshevPolynomialoftheSecondKind.html) but neither used the nomenclature "weights".

Maybe just giving a quick hint would help more users?  Just giving my impressions as someone new to the code :)

Reply:
Yeah I updated the docs. Weights should only refer to the barycentric interpolation weights. I've used Parameters in the docstring instead since they represent the coefficients.